oauthproxy.go

```diff
@@ -429,9 +429,16 @@ func (p *OAuthProxy) SetCSRFCookie(rw http.ResponseWriter, req *http.Request, va
 // ClearSessionCookie creates a cookie to unset the user's authentication cookie
 // stored in the user's session
 func (p *OAuthProxy) ClearSessionCookie(rw http.ResponseWriter, req *http.Request) {
-	cookies := p.MakeSessionCookie(req, "", time.Hour*-1, time.Now())
-	for _, clr := range cookies {
-		http.SetCookie(rw, clr)
+	reqCookies := req.Cookies()
+	var cookies []*http.Cookie
+
+	for _, c := range reqCookies {
```

Comment:
Nit: May as well inline this?

Reply:
Hm, I'm not sure what you're up to. I need the slice to support the "ugly hack"

```
	// ugly hack because default domain changed
	if p.CookieDomain == "" && len(cookies) > 0 {
		clr2 := *cookies[0]
		clr2.Domain = req.Host
		http.SetCookie(rw, &clr2)
	}
```
Could you provide some details in case I'm totally wrong?